src/ModelContextProtocol/McpEndpointExtensions.cs

```diff
@@ -1,13 +1,145 @@
 ﻿using ModelContextProtocol.Protocol.Messages;
 using ModelContextProtocol.Utils;
+using ModelContextProtocol.Utils.Json;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization.Metadata;
 
 namespace ModelContextProtocol;
 
 /// <summary>Provides extension methods for interacting with an <see cref="IMcpEndpoint"/>.</summary>
 public static class McpEndpointExtensions
 {
+    /// <summary>
+    /// Sends a JSON-RPC request and attempts to deserialize the result to <typeparamref name="TResult"/>.
+    /// </summary>
+    /// <typeparam name="TParameters">The type of the request parameters to serialize from.</typeparam>
+    /// <typeparam name="TResult">The type of the result to deserialize to.</typeparam>
+    /// <param name="endpoint">The MCP client or server instance.</param>
+    /// <param name="method">The JSON-RPC method name to invoke.</param>
+    /// <param name="parameters">Object representing the request parameters.</param>
+    /// <param name="parametersTypeInfo">The type information for request parameter serialization.</param>
+    /// <param name="resultTypeInfo">The type information for request parameter deserialization.</param>
+    /// <param name="requestId">The request id for the request.</param>
+    /// <param name="cancellationToken">A token to cancel the operation.</param>
+    /// <returns>A task that represents the asynchronous operation. The task result contains the deserialized result.</returns>
+    public static async Task<TResult> SendRequestAsync<TParameters, TResult>(
+        this IMcpEndpoint endpoint,
+        string method,
+        TParameters parameters,
+        JsonTypeInfo<TParameters> parametersTypeInfo,
+        JsonTypeInfo<TResult> resultTypeInfo,
```

Comment:
Are we at all concerned about the ergonomics of making the JsonTypeInfo's required parameters to SendRequestAsync? Or do we think most usage will be via extension methods like CallToolAsync meaning it won't matter?

If we think people will want to customize the JSON-RPC parameter and result serialization types, I could look at a wrapper method in the AspNetCore project that gets the JsonTypeInfo's from IOptions<JsonOptions> where you might have, you're AOT-generated resolvers registered in the TypeInfoResolverchain.

Reply:
> Are we at all concerned about the ergonomics of making the JsonTypeInfo's required parameters to SendRequestAsync? 

Good question. I'm exposing the JTI overload as a means to get compile-time feedback on whether the contract for a given model has been source generated. As such, it makes sense more in the context of the library implementation itself. I can mark it internal and only expose the JSO-based overload if you feel this is adding complexity to users.

Reply:
Not yet. We would need a separate test project for running AOT smoke tests. It's something we need to get to eventually.

Reply:
We could try incorporating this into @agocke's PR so that it's exercised as well.